api.yml

```diff
@@ -19,9 +19,97 @@ paths:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              description: The credential to be verified and a set of options to use when verifying the credential.
+              type: object
+              properties:
+                verifiableCredential:
+                  type: object
+                  additionalProperties: false
+                  description: The verifiable credential to check.
+                  example: {
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1"
+  ],
+  "id": "http://example.edu/credentials/3732",
+  "type": [
+    "VerifiableCredential",
+    "UniversityDegreeCredential"
+  ],
+  "issuer": "did:example:c6f1c276e12ec21ebfeb1f712eb",
+  "issuanceDate": "2020-03-11",
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "degree": {
+      "type": "BachelorDegree",
+      "name": "Bachelor of Science and Arts"
+    }
+  },
+  "proof": [{
+    "type": "Ed25519Signature2018",
+    "created": "2020-03-11T23:09:06.803Z",
+    "jws": "eyJhbGciOiJFZERTQSIsImI2..hS5SPmWTFocjcDg",
+    "proofPurpose": "assertionMethod",
+    "verificationMethod": "did:example:c6f1c276e12ec21ebfeb1f712eb#jf893k"
+  }]
+}
+                options:
+                  type: object
+                  additionalProperties: false
+                  description: Options for specifying how the credential will be issued.
+                  properties:
+                    checks:
+                      type: array
+                      items:
+                        type: string
+                      description: An array of verification checks to be performed on the credential. WARNING - Implementers are cautioned that the list of checks is currently incomplete and other checks such as issuance, expiration, nonce, domain, and acceptable issuers are expected to be added in time. A default list is also expected to encapsulate all available checks. This option might be changed to a 'ignore' option where it lists the checks the developer would like to skip.
+                      example: ["proof"]
+      responses:
+        '200':
+          description: The verification was successful.
+          content:
+            application/json:
+              schema:
+                type: object
+                description: A description of the checks that were performed.
+                properties:
+                  checks:
+                    type: array
+                    items:
+                      type: string
+                    description: An array of verification checks that were performed. WARNING - Implementers are cautioned that the list of checks is currently incomplete and other checks such as issuance, expiration, nonce, domain, and acceptable issuers are expected to be added in time.
```

Comment:
I want to register a concern here. It's not anything that needs to derail this particular PR, or other near-term progress -- but it's something we ultimately need to wrestle with and document.

What we are doing here is trusting the service to tell us whether a credential has been verified. As long as the API we're building is meant to be "internal" (that is, the service and the client of the service are both running inside a trust context governed by the same identity), trusting what the service asserts here is probably good enough. But if we ever run the service in a mode where a trust boundary is crossed (e.g., a SaaS service operated by a different entity from the client), then there is a trust issue. A way to resolve it would be to return a state proof. But of course that then creates a crypto and processing burden on the client, which is likely not to be viable in many cases. So the moral to the story, for me, is that there are subtle trust issues that we need to understand, that have not yet surfaced.

Reply:
>  if we ever run the service in a mode where a trust boundary is crossed (e.g., a SaaS service operated by a different entity from the client), then there is a trust issue. A way to resolve it would be to return a state proof.

Agreed. I've raised issue #10 to track this.

> But some kinds of verification may not be doable by a service (e.g., validate business requirements that only the client knows about).

Agreed. I've raised #11 to track this.